src/MSISonar.cc

```diff
@@ -524,7 +536,25 @@ void MSISonar::GetSonarImage(double _angleDisplacement)
 
   // this->DebugPrintMatrixToFile<int>("Teste3.dat", transferTable);
 
-  this->TransferTableToSonar(accumData, transferTable);
+  this->TransferTableToSonar(this->accumData, transferTable);
+}
+
+//////////////////////////////////////////////////
+sonar_msgs::SonarStamped MSISonar::SonarRosMsg(const gazebo::physics::WorldPtr _world, float _actAngle)
```

Comment:
I think that in cases it returns `sonar_msgs::SonarStamped` objects you could consider the use of references or pointers. A common ROS approach would consider a boost pointer in these cases, I believe that it would be similar to: `typedef boost::shared_ptr<sonar_msgs::SonarStamped> SonarStampedPtr`.

Reply:
I agree and have done this modification